insights/parsers/fcoeadm_i.py

```diff
@@ -0,0 +1,165 @@
+
+    Examples:
+        >>> type(fi)
+        <class 'insights.parsers.fcoeadm_i.FcoeadmI'>
+        >>> fi.fcoe["Description"]
+        'NetXtreme II BCM57810 10 Gigabit Ethernet'
+        >>> fi.fcoe["Driver"]
+        'bnx2x 1.712.30-0'
+        >>> fi.fcoe['Serial Number']
+        '2C44FD8F4418'
+        >>> fi.get_iface
+        ['eth8.0-fcoe', 'eth6.0-fcoe']
+        >>> fi.get_nic
+        ['eth8', 'eth6']
+        >>> fi.get_stat
+        ['Online', 'Offline']
+        >>> fi.get_stat_from_nic('eth6')
+        'Offline'
+        >>> fi.get_host_from_nic('eth8')
+        'host6'
+
+    Attributes:
+        fcoe (dict): Parse the command of '/usr/sbin/fcoeadm -i'
+        iface_lenth (int): The length of fcoe interfaces
+        driver_name (str): Driver name
+        driver_version (str): Driver version
+
+    Raises:
+        ParseException: When input content is empty or there is no parsed data.
+    """
+
+    def parse_content(self, content):
+        EMPTY = "Input content is empty"
+        BADWD = "No useful data parsed in content: '{0}'".format(content)
+
+        if not content:
+            raise SkipException(EMPTY)
+
+        if len(content) < 6 and len(content) > 0:
+            raise ParseException(BADWD)
+
+        iface_list = []
+        iface = {}
+        self.iface_lenth = None
+        self.driver_name = ''
+        self.driver_version = ''
+        self.fcoe = {}
+        self.fcoe['Interfaces'] = iface_list
+        symb = False
+
+        for line in content:
+            line = line.strip()
+
+            if 'Symbolic Name:' in line:
+                symb = True
+
+            if ':' in line:
+                key, val = [s.strip() for s in line.split(': ', 1)]
+                if not symb:
+                    self.fcoe[key] = val
+                else:
+                    iface[key] = val
+
+            if 'State:' in line:
+                iface_list.append(iface)
+                iface = {}
+
+        self.iface_lenth = len(self.fcoe['Interfaces'])
+        self.driver_name = self.fcoe['Driver'].split(' ', 1)[0]
+        self.driver_version = self.fcoe['Driver'].split(' ', 1)[-1]
+        return self.fcoe
+
+    @property
+    def get_iface(self):
+        """int: the major version of this OS."""
+        iface_list = []
+        for iface in self.fcoe['Interfaces']:
+            iface_list.append(iface['Symbolic Name'].split(' ')[-1])
+        return iface_list
+
+    @property
+    def get_nic(self):
+        """list: the nic of fcoe iface"""
+        nic_list = []
+        for nic in self.get_iface:
+            nic_list.append(nic.split('.', 1)[0])
+        return nic_list
+
+    @property
+    def get_stat(self):
+        """list: the state of fcoe ifaces"""
+        stat_list = []
+        for iface in self.fcoe['Interfaces']:
+            stat_list.append(iface['State'])
+        return stat_list
+
+    def get_stat_from_nic(self, nic):
+        """str: the fcoe state"""
```

Comment:
For method requires `paraments`, please add a corresponding description for them.